sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -444,138 +444,97 @@
     }
 
     /**
-     * Set the {@link ConfigurationSetting} to read-only with a matching {@code key}, and optional {@code label}
-     * combination.
+     * Set the {@link ConfigurationSetting} to read-only or not read-only with a matching {@code key},
+     * and optional {@code label} combination. If the {@code isReadOnly} is true or null, the setting will be set to
+     * read-only. If false, the setting will not be read-only.
      *
      * <p><strong>Code Samples</strong></p>
      *
      * <p>Set the setting to read-only with the key-label "prodDBConnection"-"westUS".</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string-Boolean}
+     *
+     * <p>Clear read-only of the setting with the key-label "prodDBConnection"-"westUS".</p>
+     *
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string-Boolean-clearReadOnly}
      *
      * @param key The key of configuration setting to set to be read-only.
-     * @param label The label of configuration setting to read-only, or optionally, null if a setting with label is
-     * desired.
+     * @param label The label of configuration setting to read-only, or optionally, null if the label is not desired
+     * in the request.
+     * @param isReadOnly The boolean value to set the setting to read-only or not read-only.
+     * If it is true or null, set the setting to read-only. If false, the setting won't set to read-only.
      * @return The {@link ConfigurationSetting} that is read-only, or an empty Mono if a key collision occurs or the
      * key is an invalid value (which will also throw HttpResponseException described below).
      * @throws IllegalArgumentException If {@code key} is {@code null}.
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<ConfigurationSetting> setReadOnly(String key, String label) {
+    public Mono<ConfigurationSetting> setReadOnly(String key, String label, Boolean isReadOnly) {
```

Comment:
Could we just use a `boolean`? `null` feels ambiguous.

Reply:
updated